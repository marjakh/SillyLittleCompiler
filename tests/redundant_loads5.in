function sneaky(f) {
  f();
}

function test1() {
  print("Test 1");
  let level1var = 10;
  function level1func() {
    level1var = 20;
  }

  print(level1var);

  // Function call invalidates the variable cache (also for level1var, since the
  // function could sneakily modify it too (e.g., if it gets passed a reference to
  // an inner function as an argument).
  sneaky(level1func);
  print(level1var);
}

test1();
