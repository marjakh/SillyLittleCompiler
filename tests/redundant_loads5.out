Test 1
10
20
